pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -141,10 +155,32 @@ func TestConfigPath(t *testing.T) {
 				SystemRegistriesConfPath:     nondefaultPath,
 			},
 			nondefaultPath,
+			false,
+		},
+		// User registries file overridden
+		{&types.SystemContext{}, userRegistriesFilePath, true},
+		// Context and user User registries file preset simultaneously
+		{&types.SystemContext{SystemRegistriesConfPath: nondefaultPath}, nondefaultPath, true},
+		// Root and user registries file overrides present simultaneously,
+		{
+			&types.SystemContext{
+				RootForImplicitAbsolutePaths: rootPrefix,
+				SystemRegistriesConfPath:     nondefaultPath,
+			},
+			nondefaultPath,
+			true,
 		},
 		// No environment expansion happens in the overridden paths
-		{&types.SystemContext{SystemRegistriesConfPath: variableReference}, variableReference},
+		{&types.SystemContext{SystemRegistriesConfPath: variableReference}, variableReference, false},
 	} {
+		os.Remove(userRegistriesFilePath)
+		if c.userfile {
+			err := os.MkdirAll(filepath.Dir(userRegistriesFilePath), os.ModePerm)
+			assert.Nil(t, err)
+
+			_, err = os.Create(userRegistriesFilePath)
```

Comment:
(Nonblocking: closing the file object explicitly would be clearer. OTOH the implementation sets a finalizer anyway, so the descriptor does not actually leak forever.)